src/cpp/exception/Exception.cpp

```diff
@@ -74,13 +74,13 @@ BadParameter::BadParameter(
 {
 }
 
-NoData::NoData(
+PreconditionNotMet::PreconditionNotMet(
         const char* message) noexcept
     : Exception(message)
 {
 }
 
-NoData::NoData(
+PreconditionNotMet::PreconditionNotMet(
         const std::string& message)
     : Exception(message)
 {
```

Comment:
The proper way of handling this would be making unit tests for the exceptions, but I find it a little bit superfluous, being the basic structures that they are, and basically all having the same (almost null) functionality.

However, I understand the requirement for complete coverage for the record. I modified one of the throws so that it uses the `std::string` constructor instead, and make the exception provide more information at the same time. We fill the coverage without having to create tests without real content.

Reply:
I'm agree with that solution, I made the same with the CorruptedFile exception.